// Copyright 2023 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package collision_checking;

import "collision_checking/geometry_shapes/eigen.proto";

// A 3D axis-aligned bounding box, which is defined by the intervals
// x in [min_corner.x, max_corner.x]
// y in [min_corner.y, max_corner.y]
// z in [min_corner.z, max_corner.z]
// Next ID: 3
message AxisAlignedBoundingBox3dProto {
  collision_checking.proto.Vector3dProto min_corner = 1;
  proto.Vector3dProto max_corner = 2;
}
